core/src/main/java/tech/tablesaw/columns/numbers/IntParser.java

```diff
@@ -25,7 +28,7 @@ public boolean canParse(String str) {
     }
     String s = str;
     try {
-      if (s.endsWith(".0")) {
+      if (!zeroDecimalAsFloat && s.endsWith(".0")) {
```

Comment:
It's a little weird that it only works with a single 0. What if it was `.00`? I know you didn't write that part, but it might be a good opportunity to clean it up

Reply:
Agreed. I have updated the PR now.

I added a function to `StringUtils` so that we can centralize the logic. I opted for a regex with a compiled pattern for performance reasons. I combined the check for zero decimal with the remove operation as it is simpler and performance will be the same.

This made the `FixedWidthReaderTest#testDataTypeDetection` test fail which is legitimate as the test data contains data with two zero decimals, so I updated the test.

Reply:
Let me know if the util method is better off in another util class or if I should create a new one.

Reply:
This seems fine. Would you mind adding the test?

Reply:
I have added tests now. These include checks for the column type detection feature and for getting columns after reading the data into a table. There are tests asserting behavior when `ignoreZeroDecimal` is disabled as well as enabled.

Let me know if the tests look appropriate.